sql_datafreshness/sql_datafreshness_compilation.py

```diff
@@ -0,0 +1,179 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#    Jesús Alonso Barrionuevo <jalonso@bitergia.com>
+
+
+import os
+import sys
+import time
+import subprocess
+import ConfigParser
+import logging
+from argparse import ArgumentParser
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+import smtplib
+
+class Record_machine(object):
+    __slots__= "name","connection", "log_from",
+
+    def items(self):
+        "dict style items"
+        return [
+            (field_name, getattr(self, field_name))
+            for field_name in self.__slots__]
+    def __set__(self,attr,value):
+	"set a value in an item"
+	setattr(self, attr,value)
+    def getattrs(self):
+	"get the list of attrs"
+	return self.__slots__
+    def __iter__(self):
+        "iterate over fields tuple/list style"
+        for field_name in self.__slots__:
+            yield getattr(self, field_name)
+    def __getitem__(self, index):
+        "tuple/list style getitem"
+        return getattr(self, self.__slots__[index])
+
+
+def get_args():
+	parser = ArgumentParser(usage='Usage: %(prog)s [options]',
+		            description='Checks data freshness log of SQL databases in all machines and made a new mail',
+		            version='0.1')
+	parser.add_argument('-g','--debug', dest='debug',
+		help='Debug mode, disabled by default',
+		required=False, action='store_true')
+	parser.add_argument('--conf', dest='config_file',
+		help='Configuration file',
+		required=False)
+	parser.add_argument('-s','--send', dest='send',
+		help='Sends the information to the mail. Disabled by default.',
+		required=False,action='store_true')
+	parser.add_argument('-f','--file', dest='file',
+		        help='Generates a file with the content of the report.',
+		        required=False,action='store_true')
+	args = parser.parse_args()
+
+	if len(sys.argv)==1:
+		parser.print_help()
+		sys.exit(1)
+
+    	return args
+
+def read_conf(file_path):
+
+	Config = ConfigParser.ConfigParser()
+	Config.read(file_path)
+
+	machines=[]
+	config_machines=[]
+	opts={}
+
+	str_machines_option =Config.get("config","machines")
+	machines=str_machines_option.split(',')
+	for machine in machines:
+		r=Record_machine()
+		for attribute in r.getattrs():
+			if attribute=="name":
+				r.name=machine
+			else:
+				r.__set__(attribute,Config.get("config",machine+"_"+attribute))
+		config_machines.append(r)
+	
+	try:
+       		opts['email_from'] = Config.get('config','email_from')
+	except:
+		opts['email_from'] = ''
+	try:
+		opts['email_to'] = Config.get('config','email_to')
+	except:
+		opts['email_to'] = ''
+	try:
+		opts['log_files'] = Config.get('config','log_files')
+	except:
+		opts['log_files'] = ''
+
```

Comment:
This code is not sexy at all. Take into account something else could happen to trigger an expection. Why don't you set a default value and assign the value if it exists?